kasa/utils.py

```diff
@@ -0,0 +1,37 @@
+"""Python library supporting TP-Link Smart Home devices.
+
+The communication protocol was reverse engineered by Lubomir Stroetmann and
+Tobias Esser in 'Reverse Engineering the TP-Link HS110':
+https://www.softscheck.com/en/reverse-engineering-tp-link-hs110/
+
+This library reuses codes and concepts of the TP-Link WiFi SmartPlug Client
+at https://github.com/softScheck/tplink-smartplug, developed by Lubomir
+Stroetmann which is licensed under the Apache License, Version 2.0.
+
+You may obtain a copy of the license at
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+
+
+import collections.abc
+from collections import defaultdict
+from typing import DefaultDict
+
+
+def merge(d, u):
```

Comment:
Considering these both are used just in a single place, I don't think it makes sense to create a separate module for storing them.

Reply:
I'll move them back into the modules where they are used as separate functions so at least we can avoid recreating them inline when the function is called.